### PR TITLE
Auto-generate and store secret keys

### PR DESCRIPTION
**Pull request by josephschorr**
Since the secret keys need to be persistant across runs, we now auto-generate and then store them in a specialized Secret

Addresses https://issues.redhat.com/browse/PROJQUAY-882

### CODE REVIEW

File: pkg/kustomize/kustomize.go [hunk 1]
```diff
@@ -20,9 +21,17 @@ import (
...
+	"github.com/go-logr/logr"
 	v1 "github.com/quay/quay-operator/api/v1"
 )
 
+const configSecretPrefix = "quay-config-secret"
+const secretKeyLength = 80
+
+// SecretKeySecretName is the name of the Secret in which generated secret keys are
+// stored.
+const SecretKeySecretName = "quay-registry-managed-secret-keys"
```
Comment (alecmerdler) [inline pkg/kustomize/kustomize.go:33]: This needs to be prefixed with the `metadata.name` of the `QuayRegistry` or else we will have name collisions.

File: pkg/kustomize/kustomize.go [hunk 2]
```diff
@@ -216,16 +225,86 @@ func flattenSecret(configBundle *corev1.Secret) (*corev1.Secret, error) {
...
+		check(err)
+
+		stringData := secretKeysSecret.StringData
+		if stringData == nil {
+			stringData = map[string]string{}
+		}
+
+		secretKeysSecret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      SecretKeySecretName,
```
Comment (alecmerdler) [inline pkg/kustomize/kustomize.go:266]: Add the `metadata.name` of the `QuayRegistry` as a prefix.

File: pkg/kustomize/kustomize.go (hunk 1 above)
Comment (josephschorr) [reply pkg/kustomize/kustomize.go]: Done

File: pkg/kustomize/kustomize.go (hunk 2 above)
Comment (josephschorr) [reply pkg/kustomize/kustomize.go]: Done

File: pkg/kustomize/kustomize.go [hunk 3]
```diff
@@ -20,9 +21,23 @@ import (
 	"sigs.k8s.io/kustomize/api/types"
 	"sigs.k8s.io/yaml"
 
+	"github.com/go-logr/logr"
```
Comment (alecmerdler) [inline pkg/kustomize/kustomize.go:24]: Fix import order.

**Pull request merged**